pwiz_tools/Skyline/FileUI/PeptideSearch/BuildPeptideSearchLibraryControl.cs

```diff
@@ -223,28 +225,151 @@ public bool FilterForDocumentPeptides
         public string[] SearchFilenames
         {
             get { return ImportPeptideSearch.SearchFilenames; }
+            private set { SetSearchFiles(value, true); }
+        }
+
+        private void SetSearchFiles(string[] searchFiles, bool async)
+        {
+            // Set new value
+            ImportPeptideSearch.SearchFilenames = searchFiles;
 
-            private set
+            // Always show sorted list of files
+            Array.Sort(ImportPeptideSearch.SearchFilenames);
+
+            var toAdd = searchFiles.ToHashSet();
+
+            // Populate the input files list
+            lock (_gridFilesLock)
             {
-                // Set new value
-                ImportPeptideSearch.SearchFilenames = value;
+                gridSearchFiles.CellValueChanged -= gridSearchFiles_CellValueChanged;
 
-                // Always show sorted list of files
-                Array.Sort(ImportPeptideSearch.SearchFilenames);
+                var existingFiles = new HashSet<string>();
+                for (var i = gridSearchFiles.RowCount - 1; i >= 0; i--)

```

Comment:
I wonder if code like this would be easier to write if the DataSource of gridSearchFiles was set to a BindingList instead of adding and removing rows.

Reply:
Yeah, the way the files list works is a little strange but I didn't want to mess with it too much. Maybe changes for a future PR.